data-products/src/main/scala/org/sunbird/analytics/audit/AssessmentScoreCorrectionJob.scala

```diff
@@ -0,0 +1,109 @@
+    val batchIds: List[String] = modelParams.getOrElse("assessment.score.correction.batches", List()).asInstanceOf[List[String]].filter(x => x.nonEmpty)
+    val isDryRunMode = modelParams.getOrElse("isDryRunMode", true).asInstanceOf[Boolean]
+    for (batchId <- batchIds) yield {
+      JobLogger.log("Started Fetching the Incorrect Max Score Value for the Batch", Option(Map("batch_id" -> batchId, "isDryRunMode" -> isDryRunMode)), INFO)
+      removeRecords(batchId = batchId, isDryRunMode = isDryRunMode)
+    }
+  }
+
+  def removeRecords(batchId: String, isDryRunMode: Boolean)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig, sc: SparkContext) = {
+    val modelParams = config.modelParams.getOrElse(Map[String, Option[AnyRef]]())
+    val outputPath = modelParams.getOrElse("csvPath", "").asInstanceOf[String]
+    // Get the Assessment Data for the specific Batch
+    val assessmentData: DataFrame = getAssessmentAggData(batchId).select("course_id", "batch_id", "content_id", "attempt_id", "user_id", "total_max_score", "total_score").persist()
+    // Take the content Id which is associated to the batch being invoked for the correction
+    val contentId: String = assessmentData.select("content_id").collect().map(_ (0)).toList.head.asInstanceOf[String]
+
+    val contentMetaURL: String = modelParams.getOrElse("contentReadAPI", "https://diksha.gov.in/api/content/v1/read/").asInstanceOf[String]
+    // Get the TotalQuestion Value from the Content Meta API
+    val totalQuestions: Int = Await.result[Int](getTotalQuestions(contentId, contentMetaURL), 60.seconds)
+    JobLogger.log("Fetched the total questions value for the processing batch", Option(Map("batch_id" -> batchId, "content_id" -> contentId, "total_questions" -> totalQuestions)), INFO)
+    val filteredAssessmentData = assessmentData.filter(col("total_max_score") =!= totalQuestions)
+      .select("course_id", "batch_id", "content_id", "attempt_id", "user_id", "total_max_score", "total_score")
+    val totalRecords = filteredAssessmentData.count()
+    JobLogger.log("Total Incorrect Records", Option(Map("total_records" -> totalRecords)), INFO)
+    if (isDryRunMode) {
+      filteredAssessmentData.select("course_id", "batch_id", "content_id", "attempt_id", "user_id", "total_max_score", "total_score")
+        .repartition(1).write.option("header", true).format("com.databricks.spark.csv").save(outputPath.concat(s"/corrected-report-$batchId-${System.currentTimeMillis()}.csv"))
+      JobLogger.log("Generated the CSV File", Option(Map("batch_id" -> batchId, "total_records" -> totalRecords), INFO))
+    } else {
+      JobLogger.log("Deleting the records from the table", Option(Map("total_records" -> totalRecords)), INFO)
+      filteredAssessmentData.select("course_id", "batch_id", "user_id", "content_id", "attempt_id").rdd.deleteFromCassandra(AppConf.getConfig("sunbird.courses.keyspace"), "assessment_aggregator")
```

Comment:
Where is the logic to revert certificates and recompute aggregates?

Reply:
1. We need to get the confirmation from Krishna to revert certificates.
2. Updating the aggregates to user_activity_agg table missed here. We will add that logic.